### PR TITLE
Closes #1778: fix locale NPE

### LINKED ISSUE #1778: Current Locale NPE crash
https://sentry.prod.mozaws.net/operations/firefox-tv/issues/5131988/?query=is:unresolved%20release:org.mozilla.tv.firefox-3.3.1

### PR DESCRIPTION
**Pull request by Baron-Severin**
### Pull Request checklist

- [ ] This PR includes thorough **tests** or an explanation of why it does not
- [X] This PR includes a **CHANGELOG entry** or does not need one
- [X] The **UI tests** are passing after this PR (`./gradlew connectedDebugAndroidTest`)
- [X] I have considered adding **QA labels** on the associated issue (not this PR; `qa-ready` or `qa-denied`)
QA-Denied.  This affects a very small number of users, it's probably not worth the time to try and reproduce.

### CODE REVIEW

File: app/src/main/java/org/mozilla/tv/firefox/components/locale/LocaleManager.java [hunk 1]
```diff
@@ -290,6 +290,11 @@ public Boolean currentLanguageIsEnglish(@NonNull Context context) {
         if (current == null) {
             current = context.getResources().getConfiguration().locale;
         }
+        // In a very small number of cases, this locale will still be null. Most of our
+        // userbase uses English as a primary language, so we default to that as a fallback
+        if (current == null) {
```
[the remark this answers is not in the document]
Comment (Baron-Severin) [reply app/src/main/java/org/mozilla/tv/firefox/components/locale/LocaleManager.java]: That was my assumption when I originally wrote this as well, but it's only true if `context.getResources().getConfiguration().locale` is guaranteed to be nonnull.  That seems not to be true a small proportion of the time.

[the remark this answers is not in the document]
Comment (psymoon) [reply app/src/main/java/org/mozilla/tv/firefox/components/locale/LocaleManager.java]: Oh I see what you're doing here now. Good call. Then you can probably combine the logic like this:
`if ((current = context.getResources().getConfiguration().locale) == null) return true;`

[the remark this answers is not in the document]
Comment (psymoon) [reply app/src/main/java/org/mozilla/tv/firefox/components/locale/LocaleManager.java]: On a related note, it looks like `context.getResources().getConfiguration().locale` is deprecated according to the API doc:
>* @deprecated Do not set or read this directly. Use {@link #getLocales()} and
> {@link #setLocales(LocaleList)}. If only the primary locale is needed,
> <code>getLocales().get(0)</code> is now the preferred accessor.

File: app/src/main/java/org/mozilla/tv/firefox/components/locale/LocaleManager.java [hunk 2]
```diff
@@ -288,7 +289,16 @@ public Boolean currentLanguageIsEnglish(@NonNull Context context) {
...
-            current = context.getResources().getConfiguration().locale;
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                current = context.getResources().getConfiguration().getLocales().get(0);
+            } else {
+                current = context.getResources().getConfiguration().locale;
+            }
+        }
+        // In a very small number of cases, this locale will still be null. Most of our
+        // userbase uses English as a primary language, so we default to that as a fallback
+        if (current == null) {
```
Comment (psymoon) [inline app/src/main/java/org/mozilla/tv/firefox/components/locale/LocaleManager.java]: Can we add a Sentry log here? I'm genuinely curious to see if how many reach here after adding deprecation check

**Pull request merged**